Building cpr should check `LIBCURL_VERSION_NUM` before using `CURLOPT_SSLKEY_BLOB`

<!-- Provide a general summary of the issue in the Title above -->
<!-- Use Markdown to highlight and format your code! -->
<!-- https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf -->
<!-- https://developers.google.com/blockly/guides/modify/contribute/write_a_good_issue -->

## Description
In the course of packaging cpr 1.8.1 for [Homebrew](https://brew.sh) (downstream PRs https://github.com/Homebrew/homebrew-core/pull/98161, https://github.com/Homebrew/homebrew-core/pull/100206), trying to build cpr 1.8.1 against macOS-provided cURL leads to errors when the system cURL is older than 7.71.0 (i.e. on macOS versions older than 12.x Monterey).

## Expected Behavior
Build should complete successfully using the macOS system curl.

## Actual Behavior
During the CMake configure step:
```
[...]
  -- Found CURL: /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/lib/libcurl.tbd (found version "7.64.1") found components: HTTP HTTPS 
  -- Curl 7.64.1 found on this system.
[...]
```

Build fails with the following error:
```
   [ 94%] Building CXX object cpr/CMakeFiles/cpr.dir/redirect.cpp.o
  cd /tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/cpr && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DCPR_ENABLE_SSL -DCPR_FORCE_USE_SYSTEM_CURL -Dcpr_EXPORTS -I/tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/include -I/tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/cpr_generated_includes -Wall -Wextra -Wpedantic -Werror -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk -fPIC -std=gnu++11 -MD -MT cpr/CMakeFiles/cpr.dir/redirect.cpp.o -MF CMakeFiles/cpr.dir/redirect.cpp.o.d -o CMakeFiles/cpr.dir/redirect.cpp.o -c /tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/cpr/redirect.cpp
  /tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/cpr/session.cpp:495:9: error: unknown type name 'curl_blob'
          curl_blob blob{};
          ^
  /tmp/cpr-20220426-23293-rmi4y7/cpr-1.8.1/cpr/session.cpp:498:41: error: use of undeclared identifier 'CURLOPT_SSLKEY_BLOB'
          curl_easy_setopt(curl_->handle, CURLOPT_SSLKEY_BLOB, &blob);
                                          ^
  2 errors generated.
  make[2]: *** [cpr/CMakeFiles/cpr.dir/session.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  make[1]: *** [cpr/CMakeFiles/cpr.dir/all] Error 2
  make: *** [all] Error 2
```

## Possible Fix
<!-- Not obligatory, but suggest a fix or reason for the bug -->
We noticed that there's already checks using [`LIBCURL_VERSION_NUM`](https://github.com/libcpr/cpr/search?q=LIBCURL_VERSION_NUM) throughout cpr. Per https://curl.se/libcurl/c/CURLOPT_SSLKEY_BLOB.html, `CURL_SSLKEY_BLOB` was introduced in cURL 7.71.0, so there should be a check on the version number before the usage of this symbol.

## Steps to Reproduce
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
1. Download cpr 1.8.1 source on a macOS machine running version Big Sur (11) or earlier (macOS comes with cURL; these OS versions have cURL versions <7.71.0).
2. Configure with CMake such that the system cURL is used. An example command line invocation is `cmake -S . -B build -DCMAKE_INSTALL_PREFIX=$(pwd)/output -DCPR_FORCE_USE_SYSTEM_CURL=ON`.
3. Verify in the CMake output that the system cURL backend is selected (path of found CURL should end in `libcurl.tbd`).
4. Run the build. An example command line invocation is `cmake --build build`. This should fail on building `session.cpp` with the error described above.

## Context
<!-- How has this bug affected you? What were you trying to accomplish? -->
This has blocked cpr 1.8.1 from being shipped on Homebrew (and, more generally, prevents cpr 1.8.1 from being used on older versions of macOS without building/using a newer cURL version).

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: cpr 1.8.1
* Where did you get it from (e.g. conan, vcpkg, master, ...): Built from source tarball
* Operating System and version: macOS 11 (Intel and ARM64) and 10.15 (Intel)
* Link to a small example: N/A (issue affects build of cpr itself)

--- a/cpr/session.cpp
+++ b/cpr/session.cpp
@@ -491,6 +491,7 @@ void Session::Impl::SetSslOptions(const SslOptions& options) {
         if (!options.key_pass.empty()) {
             curl_easy_setopt(curl_->handle, CURLOPT_KEYPASSWD, options.key_pass.c_str());
         }
+#if SUPPORT_CURLOPT_SSLKEY_BLOB
     } else if (!options.key_blob.empty()) {
         std::string key_blob(options.key_blob);
         curl_blob blob{};
@@ -503,6 +504,7 @@ void Session::Impl::SetSslOptions(const SslOptions& options) {
         if (!options.key_pass.empty()) {
             curl_easy_setopt(curl_->handle, CURLOPT_KEYPASSWD, options.key_pass.c_str());
         }
+#endif
     }
     if (!options.pinned_public_key.empty()) {
         curl_easy_setopt(curl_->handle, CURLOPT_PINNEDPUBLICKEY, options.pinned_public_key.c_str());

--- a/include/cpr/ssl_options.h
+++ b/include/cpr/ssl_options.h
@@ -59,6 +59,9 @@
 #ifndef SUPPORT_SSL_NO_REVOKE
 #define SUPPORT_SSL_NO_REVOKE __LIBCURL_VERSION_GTE(7, 44)
 #endif
+#ifndef SUPPORT_CURLOPT_SSLKEY_BLOB
+#define SUPPORT_CURLOPT_SSLKEY_BLOB __LIBCURL_VERSION_GTE(7, 71)
+#endif
 
 namespace cpr {
 
@@ -125,6 +128,7 @@ class KeyFile {
     }
 };
 
+#if SUPPORT_CURLOPT_SSLKEY_BLOB
 class KeyBlob {
   public:
     // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
@@ -142,6 +146,7 @@ class KeyBlob {
         return "PEM";
     }
 };
+#endif
 
 using PemKey = KeyFile;
 
@@ -390,7 +395,9 @@ struct SslOptions {
     std::string cert_file;
     std::string cert_type;
     std::string key_file;
+#if SUPPORT_CURLOPT_SSLKEY_BLOB
     std::string key_blob;
+#endif
     std::string key_type;
     std::string key_pass;
     std::string pinned_public_key;
@@ -430,11 +437,13 @@ struct SslOptions {
         key_type = opt.GetKeyType();
         key_pass = opt.password;
     }
+#if SUPPORT_CURLOPT_SSLKEY_BLOB
     void SetOption(const ssl::KeyBlob& opt) {
         key_blob = opt.blob;
         key_type = opt.GetKeyType();
         key_pass = opt.password;
     }
+#endif
     void SetOption(const ssl::PinnedPublicKey& opt) {
         pinned_public_key = opt.pinned_public_key;
     }

